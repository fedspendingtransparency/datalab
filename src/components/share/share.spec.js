import React from 'react';
import { act, create } from 'react-test-renderer';
import Share from './share';
import shareStyles from './share.module.scss';

describe('Share', () => {
	const title = 'Placeholder Title';
	const text = 'Placeholder Text';
	const location = {
		pathname: '/placeholder/location/',
		href: '/placeholder/location/'
	}

	let component = create();
	act(() => {
		component = create(
      <Share
        title={title}
        text={text}
        location={location}
      />
    );
	});

	const instance = component.root;

	window.open = jest.fn();
	const url = encodeURIComponent(instance.props.location.pathname);

	it('expect the snapshot to match', () => {
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('expects the dropdown to show class when the button is clicked', async () => {
		const button = instance.findAllByType('button')[0];
		const popup = instance.findByProps({ id: 'sharePopup' });
		expect(popup.props.className).not.toContain(shareStyles.show);

		act(() => {
			button.props.onClick();
		});

		expect(popup.props.className).toContain(shareStyles.show);
	});

	it('expects the Facebook button to be open the Facebook share window with the proper url', () => {
		const button = instance.findAllByType('button')[0];
		act(() => {
			button.props.onClick();
		});

		const facebookButton = instance.findByProps({ id: 'facebook-button' });

		act(() => {
			facebookButton.props.onClick({
				target: {
					id: 'facebook-button'
				}
			});
		});

		expect(window.open).toHaveBeenCalledWith(
			`https://www.facebook.com/sharer/sharer.php?u=${url}`,
			'_blank',
			'left=20,top=20,width=500,height=500,toolbar=1,resizable=0'
		);
	})

	it('expects the Twitter button to be open the Twitter share window with the proper url', () => {
		const button = instance.findAllByType('button')[0];
		act(() => {
			button.props.onClick();
		});

		const twitterButton = instance.findByProps({ id: 'twitter-button' });

		act(() => {
			twitterButton.props.onClick({
				target: {
					id: 'twitter-button'
				}
			});
		});

		const twitterText = encodeURIComponent(instance.props.twitter || instance.props.text || instance.title);

		expect(window.open).toHaveBeenCalledWith(
			`https://twitter.com/intent/tweet?text=${twitterText}&url=${url}`,
			'_blank',
			'left=20,top=20,width=500,height=500,toolbar=1,resizable=0'
		);
	})

	it('expects the Reddit button to be open the Reddit share window with the proper url', () => {
		const button = instance.findAllByType('button')[0];
		act(() => {
			button.props.onClick();
		});

		const redditButton = instance.findByProps({ id: 'reddit-button' });

		act(() => {
			redditButton.props.onClick({
				target: {
					id: 'reddit-button'
				}
			});
		});

		expect(window.open).toHaveBeenCalledWith(
			`http://www.reddit.com/submit?url=${url}`,
			'_blank',
			'left=20,top=20,width=500,height=500,toolbar=1,resizable=0'
		);
	})

	it('expects the LinkedIn button to be open the LinkedIn share window with the proper url', () => {
		const button = instance.findAllByType('button')[0];
		act(() => {
			button.props.onClick();
		});
		const linkedinButton = instance.findByProps({ id: 'linkedin-button' });

		act(() => {
			linkedinButton.props.onClick({
				target: {
					id: 'linkedin-button'
				}
			});
		});

		expect(window.open).toHaveBeenCalledWith(
			`https://www.linkedin.com/shareArticle?mini=true&url=${url}`,
			'_blank',
			'left=20,top=20,width=500,height=500,toolbar=1,resizable=0'
		);
	})
});