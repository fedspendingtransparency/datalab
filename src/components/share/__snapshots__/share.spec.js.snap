// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share expect the snapshot to match 1`] = `
<div
  className="shareContainer"
>
  <div
    className="popup newShare"
  >
    <button
      aria-hidden="true"
      className="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-34 MuiButton-text vizShareIcon"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <test-file-stub />
        <span
          className="shareText"
        >
          Share
        </span>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
    <span
      className="popuptext right newpopup "
      id="sharePopup"
    >
      <div
        className="shareButtons"
      >
        <ul>
          <li>
            <button
              className="shareLink"
              id="facebook-button"
              onClick={[Function]}
              title="Share on Facebook"
            >
              <img
                className="iconFacebook"
                id="facebook-icon"
                role="presentation"
                src="test-file-stub"
                style={
                  Object {
                    "height": "1.25rem",
                    "width": "1.25rem",
                  }
                }
              />
              <span
                className="shareButtonText"
                id="facebook-label"
              >
                Facebook
              </span>
            </button>
          </li>
          <li>
            <button
              className="shareLink"
              id="twitter-button"
              onClick={[Function]}
              title="Share on Twitter"
            >
              <img
                className="iconTwitter"
                id="twitter-icon"
                role="presentation"
                src="test-file-stub"
                style={
                  Object {
                    "height": "1.25rem",
                    "width": "1.25rem",
                  }
                }
              />
              <span
                className="shareButtonText"
                id="twitter-label"
              >
                Twitter
              </span>
            </button>
          </li>
          <li>
            <button
              className="shareLink"
              id="reddit-button"
              onClick={[Function]}
              title="Share on Reddit"
            >
              <img
                className="iconReddit"
                id="reddit-icon"
                role="presentation"
                src="test-file-stub"
                style={
                  Object {
                    "height": "1.25rem",
                    "width": "1.25rem",
                  }
                }
              />
              <span
                className="shareButtonText"
                id="reddit-label"
              >
                Reddit
              </span>
            </button>
          </li>
          <li>
            <button
              className="shareLink"
              id="linkedin-button"
              onClick={[Function]}
              title="Share on LinkedIn"
            >
              <img
                className="iconLinkedin"
                id="linkedin-icon"
                role="presentation"
                src="test-file-stub"
                style={
                  Object {
                    "height": "1.25rem",
                    "width": "1.25rem",
                  }
                }
              />
              <span
                className="shareButtonText"
                id="linkedin-label"
              >
                Linkedin
              </span>
            </button>
          </li>
          <li>
            <button
              className="shareLink"
              id="email-button"
              onClick={[Function]}
              title="Share via Email"
            >
              <img
                className="iconEnvelope"
                id="email-icon"
                role="presentation"
                src="test-file-stub"
                style={
                  Object {
                    "height": "1.25rem",
                    "width": "1.25rem",
                  }
                }
              />
              <span
                className="shareButtonText"
                id="email-label"
              >
                Email
              </span>
            </button>
          </li>
        </ul>
      </div>
    </span>
  </div>
</div>
`;
